docs/fundamentals/databases.md

```diff
@@ -0,0 +1,58 @@
+The database is periodically compacted to reduce the operational cost of accessing indivdual items. This is achieved by flattening the underlying data store for a given range of keys. Any deleted or overwritte items in that key range are removed and the surviving data is reorganized for efficiency.
+
+Geth also tracks several performance metrics for the LevelDB database that can be monitored via the metrics subsystem. These are: 
+
+| meter                | function                                                                |
+| -------------------- | ----------------------------------------------------------------------- |
+| `compTimeMeter`      | Meter for measuring the total time spent in database compaction         |
+| `compReadMeter`      | Meter for measuring the data read during compaction                     |
+| `compWriteMeter`     | Meter for measuring the data written during compaction                  |
+| `writeDelayNMeter`   | Meter for measuring the write delay number due to database compaction   |
+| `writeDelayMeter`    | Meter for measuring the write delay duration due to database compaction |
+| `diskSizeGauge`      | Gauge for tracking the size of all the levels in the database           |
+| `diskReadMeter`      | Meter for measuring the effective amount of data read                   |
+| `diskWriteMeter`     | Meter for measuring the effective amount of data written                |
+| `memCompGauge`       | Gauge for tracking the number of memory compaction                      |
+| `level0CompGauge`    | Gauge for tracking the number of table compaction in level0             |
+| `nonlevel0CompGauge` | Gauge for tracking the number of table compaction in non0 level         |
+| `seekCompGauge`      | Gauge for tracking the number of table compaction caused by read opt    |
+
+
+## Freezer/ancients
+
+Older segments of the chain are moved out of the fast LevelDB database and into a freezer database the is less performant. Nodes rarely need to access these files so IO speed is less important and the bulk of the chain data can be stored on a cheaper HDD. Once blocks pass some threshold age (128 blocks behind the head by default) the block and receipt data is flattened and saved as a raw binary blob of data along with an index entry file used for identification. This flat file format allows Geth to capitalize on the operating system's built-in memory manager and file cacher which in turn allows Geth to maximize the available memory without hogging it from other processes. This is good for overall system stability and also reduces the strain on Go's garbage collector and reduces the total amount of memory used by Geth. 
```

Comment:
Once blocks pass some threshold age (90,000 blocks behind the head by default)

Reply:
> This flat file format allows Geth to capitalize on the operating system's built-in memory manager and file cacher which in turn allows Geth to maximize the available memory without hogging it from other processes

I am not sure about this property. From my own understanding, freezer just opens a batch of files *normally* without any fancy tricks(e.g. mmap).

Reply:
I agree with what is written. We just use the filesystem, and it's up to the OS to manage the filesystem. If the os has memory available, it will use (some of) it as a backing cache for the filesystem. So yea, we let the os do it's thing.

However, I think it's a bit "wordy". I don't see the need to "sell" the flat file format - IMO it's enough to describe it.

Reply:
```suggestion
Older segments of the chain are moved out of the fast LevelDB database and into a freezer database the is less performant. Nodes rarely need to access these files so IO speed is less important and the bulk of the chain data can be stored on a cheaper HDD. Once blocks pass some threshold age (90,000 blocks behind the head by default) the block and receipt data is flattened and saved as a raw binary blob of data along with an index entry file used for identification. 
```